qiskit_machine_learning/neural_networks/estimator_qnn.py

```diff
@@ -208,23 +219,31 @@ def input_gradients(self, input_gradients: bool) -> None:
 
     def _forward_postprocess(self, num_samples: int, result: EstimatorResult) -> np.ndarray:
         """Post-processing during forward pass of the network."""
-        return np.reshape(result.values, (-1, num_samples)).T
+        return np.reshape(result, (-1, num_samples)).T
 
     def _forward(
         self, input_data: np.ndarray | None, weights: np.ndarray | None
     ) -> np.ndarray | None:
         """Forward pass of the neural network."""
         parameter_values_, num_samples = self._preprocess_forward(input_data, weights)
-        job = self.estimator.run(
-            [self._circuit] * num_samples * self.output_shape[0],
-            [op for op in self._observables for _ in range(num_samples)],
-            np.tile(parameter_values_, (self.output_shape[0], 1)),
-        )
-        try:
-            results = job.result()
-        except Exception as exc:
-            raise QiskitMachineLearningError("Estimator job failed.") from exc
 
+        if isinstance(self.estimator, BaseEstimatorV1):
+            job = self.estimator.run(
+                [self._circuit] * num_samples * self.output_shape[0],
+                [op for op in self._observables for _ in range(num_samples)],
+                np.tile(parameter_values_, (self.output_shape[0], 1)),
+            )
+            results = job.result().values
+        elif isinstance(self.estimator, BaseEstimatorV2):
+            PUBs = []
+            for _ in range(num_samples):
+                for observable in self._observables:
+                    PUBs.append((self._circuit, [observable], parameter_values_))
+            job = self.estimator.run(PUBs, precision=0.001)
+            results = job.result()
+            results = [result.data.evs[0] for result in results]
+        else:
+            raise QiskitMachineLearningError("Wrong Estimator Type.")
```

Comment:
```suggestion
            raise QiskitMachineLearningError(f"The accepted estimators are BaseEstimatorV1 (deprecated) and BaseEstimatorV2; got {type(self.estimator)} instead.")
```

Here we need to make sure we keep two things consistent:
- `self.estimator` (here) vs `self._estimator` (in other files)
- `QiskitMachineLearningError` (here) vs `AlgorithmsError` (other files) for the same error message. In this case, I would prefer going for the first, and gradually replace the `AlgorithmsError` calls.